### PR TITLE
Basic set creation page

### PR DESCRIPTION
**Pull request by AlexNisnevich**
#912.

- `NewSet.tsx` is basically a copy of `Deck.js` with some things tweaked to be set-specific.
- `SAVE_SET` action and corresponding reducer.
- Some other stuff fixed to make saving sets and displaying saved sets work correctly.

### CODE REVIEW

File: src/common/containers/NewSet.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+type NewSetProps = NewSetStateProps & NewSetDispatchProps & { history: History };
+
+interface NewSetState {
+  filters: {
+    robots: boolean
+    events: boolean
+    structures: boolean
+  }
+  costRange: [number, number]
+  sortCriteria: 0 | 1 | 2 | 3 | 4 | 5 | 6
```
Comment (jacobnisnevich) [inline src/common/containers/NewSet.tsx]: This feels like it should be an `enum`

File: src/common/containers/NewSet.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                onRemoveCard={this.handleRemoveCard}
+                onSave={this.handleClickSaveSet} />
+            </Paper>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private renderSidebarControls = (): JSX.Element => (
```
Comment (jacobnisnevich) [inline src/common/containers/NewSet.tsx]: Separate React component?

File: src/common/containers/NewSet.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  public render(): JSX.Element {
+    const { setBeingEdited, user } = this.props;
+    const { layout, selectedCardIds } = this.state;
+
+    return (
+      <div>
+        <Helmet title={setBeingEdited ? "Editing Set" : "Creating Set"} />
+
+        <div style={{
+          display: 'flex',
```
Comment (jacobnisnevich) [inline src/common/containers/NewSet.tsx]: If this component is a model for future Typescript React components, lets pull out the styles into a separate object and use `withStyles`

File: src/common/containers/NewSet.tsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  private handleRemoveCard = (id: string) => {
+    this.setState((state) => ({
+      selectedCardIds: state.selectedCardIds.filter((cardId) => cardId !== id)
+    }));
+  }
+
+  private handleClickSaveSet = (id: string | null, name: string, cardIds: string[]) => {
+    const { allCards, user, onSaveSet, history } = this.props;
+
+    if (!user) {
```
Comment (jacobnisnevich) [inline src/common/containers/NewSet.tsx]: Why do we let people even get to this function if they're not logged in? `loggedIn` is passed in to ActiveDeck after all

File: src/common/containers/NewSet.tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        onSetCriteria={this.set('sortCriteria')}
+        onSetOrder={this.set('sortOrder')} />
+
+      <FilterControls
+        onToggleFilter={this.toggleFilter}
+        onSetCostRange={this.set('costRange')} />
+    </Paper>
+  )
+
+  // this.set(key)(value) = this.setState({key: value})
```
Comment (jacobnisnevich) [inline src/common/containers/NewSet.tsx]: This comment needed?

File: src/common/containers/Sets.tsx [hunk 6]
```diff
@@ -41,6 +47,12 @@ class SetSummary extends React.Component<{ set: w.Set }> {
 }
 
 class Sets extends React.Component<SetsProps> {
+  public static styles = {
+    newSetButtonLabel: {
+      fontFamily: 'Carter One'
```
Comment (jacobnisnevich) [inline src/common/containers/Sets.tsx]: Let's add a constant like `const WORDBOTS_FONT = 'Carter One';` or something.

File: src/common/util/firebase.ts [hunk 7]
```diff
@@ -226,11 +215,12 @@ export function saveRecentCard(card: w.Card): void {
 }
 
 export function saveSet(set: w.Set): void {
+  console.log(set);
   fb.database()
     .ref(`sets/${set.id}`)
     .update(set);  // see firebaseRules.json - this save will only succeed if either:
-                   //   (i) there is no set yet with the given id
-                   //   (ii) the set with the given id is yet unpublished and was created by the logged-in user
+                 //   (i) there is no set yet with the given id
```
Comment (jacobnisnevich) [inline src/common/util/firebase.ts]: ?

File: src/common/containers/NewSet.tsx (hunk 4 above)
Comment (AlexNisnevich) [reply src/common/containers/NewSet.tsx]: Yeah, but we have to still typecheck that `user` isn't `null`. This is just for TypeScript's sake.

File: src/common/containers/Sets.tsx (hunk 6 above)
Comment (AlexNisnevich) [reply src/common/containers/Sets.tsx]: Let's do that in a separate PR – we use Carter One in a LOT of places ...

**Pull request merged**